api/provider/aws/deploy_create.go

```diff
@@ -0,0 +1,102 @@
+package aws
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/quintilesims/layer0/common/models"
+)
+
+func (d *DeployProvider) Create(req models.CreateDeployRequest) (*models.Deploy, error) {
+	// TODO: see Line 59. Which entity ID generation pattern do we prefer? ARN to ID or the
+	// example below?
+	deployID := generateEntityID(req.DeployName)
+
+	if err := req.Validate(); err != nil {
```

Comment:
request validation happens in the `controller` layer

Reply:
✅